lib/src/components/NxToast/__tests__/NxToast.test.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React from 'react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+import { fireEvent, screen } from '@testing-library/react';
+
+import userEvent from '@testing-library/user-event';
+
+import NxToast from '../NxToast';
+import { NxToastProps } from '../types';
+import { NxErrorAlert } from '../../NxAlert/NxAlert';
+
+describe('NxToast', function() {
+  const simpleProps: NxToastProps = {
+    onClose: () => {},
+    children: <NxErrorAlert>Toast Message</NxErrorAlert>
+  };
+
+  const quickRender = rtlRender(NxToast, simpleProps),
+      renderEl = rtlRenderElement(NxToast, simpleProps);
+
+  it('renders a div with a class .nx-toast', function() {
+    const component = quickRender();
+    const toast = component.container.children[0];
+
+    expect(toast.nodeName).toEqual('DIV');
+    expect(toast).toHaveClass('nx-toast');
+  });
+
+  it('contains an NxErrorAlert with the NxToast message', function() {
```

Comment:
Actually, I'm wondering if we should avoid the alert aspects entirely within this test.  The fact is that different kinds of `NxAlert`s use different roles by default, and we'd rather avoid testing all of them.  So maybe what we should do instead is define a `MockAlert` component within this test file which just renders a div with a particular `data-testid`, a simple close button (for testing the `onClose` wiring), and its child text content.  Then all the `NxToast` tests can use that rather than worrying about which kind of real `NxAlert` might be used.

Reply:
> After re-reading the exact wording we used to document `NxToast`'s `children` prop I've changed my mind. We should keep it to `NxAlert` but I still think we should try to avoid the details of which kind of alert it is, so let's use the bare `NxAlert` component and not `NxErrorAlert`. We can just assert the text content of the toast overall rather than the element with the `alert` role.

Changed to an NxAlert here: [75abd56](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/75abd5697e903458941a7c13e4ffe9fc2d4b330c). In order to use a bare NxAlert, it required an icon, so it has been imported into the test as well.